submissions/YuraZagor/tjsw/index.js

```diff
@@ -0,0 +1,154 @@
+const manNamesArr = ['Vasil', 'Yurij', 'Zahar', 'Taras', 'Olexiy', 'Oleg', 'Mykola', 'Sashko', 'Sergij', 'Bohdan']
+const womanNamesArr = ['Hanna', 'Helena', 'Vasilisa', 'Yulya', 'Oksana', 'Olena', 'Olga', 'Myroslava', 'Svitlana', 'Bohdana']
+const animalNamesArr = ['Joy', 'Happy', 'Lucky', 'Puffy']
+
+let catSay = 'meau';
+let dogSay = 'woufff';
+let humanSay = 'Glory to Ukraine!';
+
+function randomSelect(arr){
+  return arr[Math.floor(Math.random()*arr.length)]
+}
+let manName = randomSelect(manNamesArr)
+let womanName = randomSelect(womanNamesArr)
+let animalName = randomSelect(animalNamesArr)
+
+const inhabitants = [
+  { species: 'human', 
+    gender: 'male',
+    name: 'Vasil',
+    saying: 'Slava natsiyi',
+    friend: 'Hanna',
+    legs: 2,
+    hands: 2,
+  },
+  { species: 'human', 
+    gender: 'female',
+    name: 'Hanna',
+    saying: 'Smert vorogam',
+    friend: 'Vasil',
+    legs: 2,
+    hands: 2,
+  },
+];
+
+class Inhabitant {  
+  constructor(species, name, saying, friend, gender, habitat=inhabitants){
+    this.species=species;
+    this.name=name;
+    this.saying = saying
+    this.friend=friend;
+    this.gender=gender;
+    this.habitat = habitat
+  }
+  accountInhabitants(){
+    this.habitat.push(this);
+  }
+  commonProperties(){
+    const { species, name, saying, friend, gender} = this
+    const commonProperties = [species, name, saying, friend, gender]
+    return commonProperties    
+  }
+};
+
+class Animal extends Inhabitant{
+  constructor( species, name, saying, friend, gender, paws, tail ){
+    super(species, name, saying, friend, gender);
+    this.paws=paws;
+    this.tail=tail;
+  };
+  populate(){
+    const { paws, tail} = this
+    const speciesProperties = [ paws, tail]
+    print(this.commonProperties().concat(speciesProperties).join('; '),'div');
+    this.accountInhabitants()
+  };
+};
+
+class Dog extends Animal{
+  constructor( name=randomSelect(animalNamesArr), friend=randomSelect(inhabitants).name, gender='male' ){
+    super('dog', name, dogSay, friend, gender, 4, 1);
+  };
+};
+class Cat extends Animal{
+  constructor( name=randomSelect(animalNamesArr), friend=randomSelect(inhabitants).name, gender='female' ){
+    super('cat', name, catSay, friend, gender, 4, 1);
+  };
+};
+
+class Human extends Inhabitant{
+  constructor( name=randomSelect(womanNamesArr), friend=randomSelect(inhabitants).name, gender='female', legs=2, hands=2){
```

Comment:
Any good reason to default `gender` to `female`?

Reply:
nope, it could ass well be gender : 'male' + name from menNamesArr   combination, just wanted to awoid name/gender confusions when using uncontrolled creation

Reply:
Not the right place in the code to solve this problem. It is not the class' fault that someone may get confused. It's a developer's responsibility to minimize the confusion.